clang-tools-extra/docs/clang-tidy/checks/daedalean-operator-overloading.rst

```diff
@@ -0,0 +1,17 @@
+.. title:: clang-tidy - daedalean-operator-overloading
+
+daedalean-operator-overloading
+==============================
+
+Daedalean coding standards for operator overloading
+ - Operators , && || MUST not be overloaded.
+ - Unary operator & MUST not be overloaded.
+ - If operator[] overloaded with non-const version, const version MUST be present.
+ - Assignment operators MUST return reference to self.
+ - Assignment operators MUST check if self-assignment takes place.
```

Comment:
the self-assignment is not checked by this rule, is it?

Reply:
There is another check for self-assignment.

Reply:
OK cool, and you removed the line from the documentation for this check :+1: